[permissions] Revert default behaviour for explicit permissions (#54419)
sourcegraph/sourcegraph: doc/admin/permissions/api.md

After you enable the permissions API, you must [set permissions](#setting-repository-permissions-for-users) to allow users to view repositories (site admins bypass all permissions checks and can always view all repositories).

> > NOTE: Enabling the permissions API marks repositories as private by default, allowing you to manage access to all repositories manually. If, however, you would like to make all repositories for a specific code host connection Public, then you can set `"enforcePermissions": false` in that code host connection's configuration.

Author: Well, it's more like "unset" by default. We can mention somewhere else when you'd want to set it to true, but with the default behaviour it depends on other settings. `enforcePermissions` acts more like an override setting

> > NOTE: Enabling the permissions API marks repositories as private by default, allowing you to manage access to all repositories manually. If, however, you would like to make all repositories for a specific code host connection Public, then you can set `"enforcePermissions": false` in that code host connection's configuration.

Reviewer: We have two `> NOTE:` right after reach other, leading to a lot of visual noise.

In that case, I think it might be better to drop the `> NOTE:` and instead make the important bits of each paragraph *bold*:

```suggestion
Enabling the permissions API *marks repositories of the code host connection as private by default*. If you want to make the repositories public, set `"enforcePermissions": false` in the code host connection's configuration.
```
